### PR TITLE
docs: upgrade storybook to v7

### PR DESCRIPTION
**Pull request by petermakowski**
## Done

- docs: upgrade storybook to v7
- remove deprecated `defaultValue` from stories https://storybook.js.org/docs/react/api/arg-types#defaultvalue

### Storybook

To see rendered examples of all react-components, run:

```shell
yarn start
```

### QA in your project

from `react-components` run:

```shell
yarn build
npm pack
```

Install the resulting tarball in your project with:

```shell
yarn add <path-to-tarball>
```

### QA steps

- Run yarn docs locally and verify docs are displayed correctly

## Fixes

Fixes: # .

## Screenshots
Google Chrome screenshot 001089@2x

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -99,7 +102,11 @@
   "resolutions": {
     "@types/react": "18.2.28",
     "@types/react-dom": "18.2.13",
-    "postcss": "^8.3.11"
+    "postcss": "^8.3.11",
+    "string-width": "^4",
```
Comment (bartaz) [inline package.json:106]: Why do we need more of these resolutions? Are there some dependency incompatibilities?

Comment (petermakowski) [reply package.json:106]: https://github.com/storybookjs/storybook/issues/22431#issuecomment-1630086092

**Comment by petermakowski** [comment]
All issues have been resolved and this is ready for review now @bartaz

**Comment by bartaz** [comment]
There seem to be some breaking changes in the styling that percy surfaces:
https://percy.io/bb49709b/react-components/builds/31116305/

**Comment by bartaz** [comment]
Seems like this may be caused by this bug: https://github.com/storybookjs/storybook/issues/18921
There are some workarounds mentioned in the comments.

**Comment by petermakowski** [comment]
> Seems like this may be caused by this bug: [storybookjs/storybook#18921](https://github.com/storybookjs/storybook/issues/18921) There are some workarounds mentioned in the comments.

@bartaz This seems to be correct as in mdx files each new line produces a new paragraph. I'll update this PR to conform with mdx interleaving rule https://mdxjs.com/docs/what-is-mdx/#interleaving

Long term solution for this would be to move over to the new CSF story format https://storybook.js.org/docs/react/api/csf

**Comment by petermakowski** [comment]
@bartaz I fixed the failing stories manually and created an issue to follow this up and use the latest storybook story format: https://github.com/canonical/react-components/issues/1001

**Comment by petermakowski** [comment]
Seems that `defaultValue` has been deprecated the in latest storybook as well, I had to make more changes to fix components docs that weren't displayed with intended values. https://storybook.js.org/docs/react/api/arg-types#defaultvalue

File: renovate.json [hunk 2]
```diff
@@ -1,3 +1,3 @@
 {
-  "extends": ["github>canonical/renovate-apps"]
+  "extends": ["github>canonical/vanilla-framework"]
```
Comment (bartaz) [inline renovate.json:2]: @petermakowski Why changing this? I thought that renovate-apps was our standard for React dependencies?

File: src/components/EmptyState/EmptyState.stories.mdx [hunk 3]
```diff
@@ -33,32 +47,34 @@ export const doNothing = () => {};
...
-        <a href="#todo" target="_blank">
-          How to create instances
-          <Icon name={ICONS.externalLink}/>
-        </a>
-      </p>
-      <Button appearance="positive" onClick={doNothing}>
-        Create instance
-      </Button>
-    </EmptyState>
+    {() => (
```
Comment (bartaz) [inline src/components/EmptyState/EmptyState.stories.mdx:50]: Why story contents need to be returned from function rather then being direct children?

**Review by bartaz — APPROVED** [review]
That's great! Couple clarifying questions in comments but nothing blocking.

File: renovate.json (hunk 2 above)
Comment (huwshimi) [reply renovate.json:2]: Also, the Vanilla repo doesn't contain an extendable config (see https://github.com/canonical/react-components/pull/997).

File: src/components/EmptyState/EmptyState.stories.mdx (hunk 3 above)
Comment (petermakowski) [reply src/components/EmptyState/EmptyState.stories.mdx:50]: Temporary workaround for mdx new lines we discussed before (extra paragraph wrapper). https://github.com/canonical/react-components/issues/1001

**Review by edlerd — APPROVED** [review]
Looks good! Thank you for keeping the version up to date and fixing all the doc files to be compatible.

**Pull request merged**